agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/StatsbeatModule.java

```diff
@@ -101,6 +140,37 @@ public NetworkStatsbeat getNetworkStatsbeat() {
     return networkStatsbeat;
   }
 
+  public FeatureStatsbeat getInstrumentationStatsbeat() {
+    return instrumentationStatsbeat;
+  }
+
+  // send network statsbeat whenever redirect happens since url has been changed.
+  // new url is always retrieved from the redirect policy cache map and we don't update the
+  // endpoint.
+  public void sendNetworkStatsbeatOnRedirect(String redirectUrl) {
+    if (!disabledAll.get()) {
+      networkStatsbeat.trackHostOnRedirect(telemetryClient, redirectUrl);
```

Comment:
`telemetryClient` is going to be `null` here if `config.internal.statsbeat.disabled` is true and `config.internal.statsbeat.disabledAll` is false (related to comment above)